File: book_stores_scrapper/spiders/czytam-pl.py
import scrapy

class CzytamPlSpider(scrapy.Spider):
    name = "czytam-pl"
    start_urls = [
        'https://czytam.pl'
    ]

    #def parse(self, response):
    #   filename = 'czytampl-item.html'
    #   with open(filename, 'wb') as f:
    #      f.write(response.body)
    def parse(self, response):
        # follow links to lists pages
        for a in response.css('li.header-nav-list div div ul li a'):
            yield response.follow(a, self.parse_list)
        
    def parse_list(self, response):
        # follow links to items pages
        for product in response.css('div.product-info h3 a'):
            yield response.follow(product, self.parse_book)

        # follow pagination links
        for page in response.css('div.pagination li a'):
            yield response.follow(page, self.parse_list)

    def parse_book(self, response):
        for div in response.css('div.productreader'):
            yield {
                "title": div.css('div.show-for-medium-up h1::text').extract_first(),
                "author": div.css('div.show-for-medium-up h2.headline-azure a strong::text').extract_first(),
                "price": div.css('div.price strong::text').extract_first(),
                "currency": div.css('div.price strong span::text').extract_first(),
                "params": div.xpath(".//ul[contains(@class, 'parametry')]/li").xpath('normalize-space(string(.))').extract()
            }

File: Scrapper/Spiders/CzytamPl.cs
using IronWebScraper;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TeltIt.BSG.DataModel;
using TeltIt.BSG.Scrapper.Readers;

namespace TeltIt.BSG.Scrapper.Spiders
{
    class CzytamPl : Spider
    {
        public override void Init()
        {
            this.LoggingLevel = WebScraper.LogLevel.All;

            // Initiate the reader
            reader = new CzytamPlReader();

            // Read the sitemap (currently known 15 pages)
            HashSet<string> urls = new HashSet<string>();
            for (int i = 1; i <= 1; i++)
            {
                //urls.UnionWith(SitemapReader.ReadFromLoc(String.Format("https://czytam.pl/sitemap/mapa{0}.xml", i)));
            }

            // Parse each page find in the sitemap
            //foreach (string pageUrl in urls)
            //    this.Request(pageUrl, Parse);
            this.Request("https://czytam.pl/k,ks_777344,Porozmawiajmy-jak-dorosli-Warufakis-Janis.html", Parse);
            //this.Request("https://czytam.pl/k,ks_619022,Post-Daniela-Dajka-Krystyna-Piorkowski-Lukasz.html", Parse);
        }

        public override void Parse(Response response)
        {
            // Give response to the reader
            reader.Response = response;

            // Determine tyoe of the item
            var item = reader.ReadItem();
        }

        

        private void ParseList(Response response)
        {
            // follow links to items pages
            foreach (var product in response.Css("div.product-info h3 a"))
            {
                this.Request(product.Attributes["href"], ParseProduct);
            }

            // @TODO: add pagination
        }

        private void ParseProduct(Response response)
        {
            var divProduct = response.Css("div.productreader");

            var title = divProduct.CSS("div.show-for-medium-up h1")[0].TextContentClean;
            Scrape(new ScrapedData() { { "Title", title } });
        }

        
    }
}
